Modules/_datetimemodule.c

```diff
@@ -4848,9 +4889,17 @@ datetime_fromisoformat(PyObject* cls, PyObject *dtstr) {
         return NULL;
     }
 
-    Py_ssize_t len;
+    int needs_decref = 0;
+    dtstr = _sanitize_isoformat_str(dtstr, &needs_decref);
```

Comment:
`dtstr` should be checked for NULL.

Reply:
Good catch, C programming is hard. :(

Fixed now.